### DISCUSSION TITLE
Could I remove or insert documents while iterating the db?

### DISCUSSION THREAD

**Discussion by hao-lee** [msiemens/tinydb#484] [Q&A] [answered]
Such as
```
for item in db:
    if match(item):
        q = Query()
        db.remove(q.name == item["name"])
        new = construct_new(item)
        db.insert(new)
```

Python doesn't allow `list` to be manipulated like this. I don't know how tinydb handle this situation. Thanks.

**Comment by msiemens** [comment, marked as answer]
Hey @hao-lee, this is indeed allowed as TinyDB first reads the storage and then returns an iterator over all documents. The iterator will continue yielding the original set of documents, though. This means that any changes performed while iterating the list will not be represented in subsequent iterations.
